### ISSUE TITLE
Cannot use <ctime> instead of <time.h>

### ISSUE THREAD

**Issue by koegl** [include-what-you-use/include-what-you-use#1428]
I don't want to use C-headers, so I wanted to map `<time.h>` to `<ctime>` in my `mapping.imp` file like so:

`
[
  { include: ["<time.h>", "private", "<ctime>", "public"] }
]
`

However, this fails with the following error:
`iwyu_include_picker.cc:1330: Assertion failed: (*map)[key] == visibility Same file seen with two different visibilities: <time.h> Old vis: 1 New vis: 2`

How can I achieve this mapping? Is there some kind of default mapping for `<time.h>` that cannot be overwritten?

**Comment by kimgr** [comment]
Yes, unfortunately. I've started trying to loosen the constraints there so it's possible to override existing mappings, so hopefully I can get something workable soon.

**Comment by kimgr** [comment]
The only workaround I'm aware of is to use `-Xiwyu --no_default_mappings -Xiwyu --mapping_file=redacted_defaults.imp` where `redacted_defaults.imp` is a copy of the default mappings from the transitive closure of `iwyu.gcc.imp` with the relevant parts edited.

**Comment by jgollenz** [comment]
> It's even replacing an explicit `#include <cstdlib>` with `#include <stdlib.h>`

Ok, this was due to having `exit` instead of `std::exit`, my bad. I'll check if my other issues are related.

**Comment by bolshakov-a** [comment]
`<c*>` headers are suggested in C++ mode after 146292c5224f204166bb855ba8bf392685f92691. @kimgr, the bug test fails after the following changes:
```diff
--- a/tests/bugs/1428/1428.cc
+++ b/tests/bugs/1428/1428.cc
@@ -13,6 +13,7 @@
 #include <time.h>
 
 void f() {
+  // IWYU: time(long *) is...*<ctime>
   (void)time(nullptr);
 }
 
@@ -22,7 +23,7 @@ tests/bugs/1428/1428.cc should add these lines:
 #include <ctime>
 
 tests/bugs/1428/1428.cc should remove these lines:
-#include <time.h>  // lines XX-XX
+- #include <time.h>  // lines XX-XX
```

**Comment by kimgr** [comment]
@bolshakov-a Thanks, it's a pity I got some bug tests wrong so they fail for the wrong reasons. But let's get rid of that test and close this issue, then?

**Comment by kimgr** [comment]
I'll push a PR to get rid of the 1428 bug test.

**Comment by kimgr** [comment]
https://github.com/include-what-you-use/include-what-you-use/pull/2094

**Comment by kimgr** [comment]
This is fixed as of https://github.com/include-what-you-use/include-what-you-use/commit/146292c5224f204166bb855ba8bf392685f92691.

We still have challenges with overriding internal mappings with external .imp mappings as per the bug report, but that's tracked in #1271.
